3953 move plone app workflow to plone app layout

MrTango opened the pull request:
needs: 

- https://github.com/plone/plone.app.workflow/pull/58
- https://github.com/plone/Products.CMFPlone/pull/4336

MrTango committed:
move UI part of sharing view here

MrTango committed:
fix linting

petschki reviewed (approved):
news entry missing, otherwise LGTM

ale-rt commented on plone/app/layout/profiles/default/browserlayer.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layers>
+  <layer interface="plone.app.layout.interfaces.IBrowserLayer"
```
Better for greppability:
```suggestion
  <layer interface="plone.app.layout.interfaces.IPloneAppLayoutBrowserLayer"
```

ale-rt commented on plone/app/layout/profiles/uninstall/browserlayer.xml:
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layers>
+  <layer interface="plone.app.layout.interfaces.IBrowserLayer"
```
```suggestion
  <layer interface="plone.app.layout.interfaces.IPloneAppLayoutBrowserLayer"
```

ale-rt commented on plone/app/layout/views/configure.zcml:
```diff
@@ -1,7 +1,19 @@
+      for="*"
+      class=".sharing.SharingView"
+      permission="plone.DelegateRoles"
+      layer="plone.app.layout.interfaces.IClassicUILayer"
```
```suggestion
      layer="plone.app.layout.interfaces.IPloneAppLayoutBrowserLayer"
```

MrTango commented:
i added a separate PR #391 for the browserlayer

MrTango committed:
add changelog entry

MrTango committed:
add test_sharing_view

MrTango committed:
fix linting

MrTango committed:
fix import of message factory

MrTango committed:
fix BrowserLayer name

MrTango committed:
fix test setup

ale-rt commented on plone/app/layout/profiles/uninstall/browserlayer.xml:
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <layers>
-  <layer name="plone.app.layout"
+  <layer interface="plone.app.layout.interfaces.IPloneAppLayoutLayer"
+         name="plone.app.layout"
```
```suggestion
  <layer name="plone.app.layout"
```

ale-rt commented on plone/app/layout/views/configure.zcml:
```diff
@@ -1,7 +1,19 @@
 
+  <include
+      package="plone.app.workflow"
+      file="permissions.zcml"
```
Q: should all the package be included?

MrTango committed:
rm IPloneAppLayoutLayer as suggested by ale

petschki commented on src/plone/app/layout/profiles/uninstall/browserlayer.xml:
maybe creating a small PR with this change would be better than inside this PR?

MrTango committed:
Add sharing view with CSRF doctest moved from Products.CMFPlone

- Fix PloneMessageFactory import in controlpanels/sharing.py (use plone.base).
- Fix testing-layer name in test_sharing_view.py.
- Add sharing_csrf.txt functional doctest under the layout functional layer,
  which installs plone.app.layout:default so IPloneAppLayoutLayer is active.

MrTango committed:
Merge branch 'master' into 3953-move-plone-app-workflow-to-plone-app-layout

MrTango committed:
Fix isort import ordering in sharing controlpanel and CSRF test
